public-awesome/stargaze
# feat: integrate v4 of the packet forward middleware

## Description 

This PR integrates `v4` of the `packet-forward-middleware` from the [ibc-apps repo](https://github.com/cosmos/ibc-apps/tree/release/v4/middleware/packet-forward-middleware). A summary of the integration path and some of the configurable options in the PFM can be found [here](https://github.com/cosmos/ibc-apps/blob/release/v4/middleware/packet-forward-middleware/docs/integration.md).

## Testing

To assert that the PFM is working as intended I have included an e2e test which can be found in the directory `e2e`.  The e2e test utilizes [interchaintest](https://github.com/strangelove-ventures/interchaintest) and makes use of a locally built Docker image. I have included some Make commands to simplify setting up [heighliner](https://github.com/strangelove-ventures/heighliner), building the Docker image and running the tests.

To run the test with the changes in this PR follow the directions below.

1. Clone the forked repo and checkout my feature branch

```bash
git clone https://github.com/jtieri/stargaze.git && git checkout feat/packet-forward-middleware
```

2. Build the local Docker image, this requires installing [heighliner](https://github.com/strangelove-ventures/heighliner)

```bash
make get-heighliner && make local-image
```

3. Run the e2e test

```bash
make test-e2e
```

The testing setup could be changed in any way you see fit and has mostly been included here as a means for verifying that the integration of the PFM is successful. Feel free to edit or completely remove this component of the PR!
